CloudWatch: Run query on blur in logs query field

fridgepoet opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

#47224 noted that queries should be run whenever the `MultiSelect` or `QueryField` for `LogsQueryField` are blurred.

**Which issue(s) this PR fixes**:
<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #47224

**Special notes for your reviewer**:
- Shortcut to run just the cloudwatch tests: `yarn jest -i --testPathPattern=/cloudwatch/components/`
- Create a CloudWatch panel and specify at least one log group along with a query and note whenever either field is blurred the "refresh dashboard" icon spins to indicate the query is being run.

fridgepoet committed:
cloud-datasources mob! :snail:

lastFile:public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

sarahzinger committed:
cloud-datasources mob! :sob:

iwysiu committed:
cloud-datasources mob! :snail:

asimpson committed:
cloud-datasources mob! :baby:

fridgepoet committed:
cloud-datasources mob! :snail:

lastFile:public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=47454) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=47454) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: fridgepoet<br/>:white_check_mark: sarahzinger<br/>:white_check_mark: iwysiu<br/>:x: asimpson<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=47454) it.</sub>

asimpson committed:
Cloudwatch: add test to ensure RunQuery is fired onBlur of LogQuery

Also remove the `onBlur` prop for the `QueryField` in `LogsQueryField` since `QueryField` is configured to re-run
queries `onBlur` by default.

iwysiu commented on public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx:
```diff
@@ -344,7 +345,6 @@ export class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogs
               additionalPlugins={this.plugins}
               query={query.expression ?? ''}
               onChange={this.onChangeQuery}
-              onBlur={this.props.onBlur}
```
Should the query field also run on blur? (Assuming that there's at least one selectedLogGroup)

asimpson replied:
@sarahzinger @iwysiu `QueryField` is configured by default to run the query on blur: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/QueryField/QueryField.tsx#L189 Passing the prop would be needed if we wanted different behavior.

iwysiu replied:
👍  thanks for clarifying

sarahzinger commented on public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx:
```diff
@@ -6,12 +6,40 @@ import { DescribeLogGroupsRequest } from '../types';
+          {
+            getVariables: () => [],
+            getRegions: () => Promise.resolve([]),
+          } as any
```
I believe we have some sort of function that's like setupMockDatasource or something like that, might be nice to use that right?

sarahzinger reviewed (approved):
Looks good to me, I have the same q as isabella, I think if we update either the log group or the query editor I would expect we'd run on query on blur I think.
